src/annotations/deprecated.cr

```diff
@@ -0,0 +1,11 @@
+# This annotations marks methods, types or constants as deprecated.
```

Comment:
Only methods

Reply:
The compiler `--check` feature only supports methods, but the annotation can be used on anything, right?

Reply:
That should be right. There is no check whether the annotation makes sense unless it is used in the semantic or codegen phase.

Reply:
Also, the `--check` is not the final option. I need to update the PR description...

Reply:
I'd like to view this annotation independent of the specific compiler feature. `--check` is just one application where it is used. The doc generator would be another one. And I could easily see both supporting types and constants as well in the future.

Reply:
Yeah, better rename it. I'm aware that it has changed, but `--check` feels like a good way to address it... :P

Reply:
I can imagine people using `Deprecated` on types and opening an issue here because it doesn't work.

Reply:
And PRs will be welcomed